### PR TITLE
Issue2: Adding simple User REST API service

### PR DESCRIPTION
**Pull request by denkonakov**
WIP: for the issue #2 

So far has been done:

1)  Created simple V1 (in memory) controller for UserWithRoles list
2)  Trivial CRUD (without UPDATE) supported
3)  UT for the CRUD (without UPDATE) has been added

### CODE REVIEW

File: api-server/src/main/java/org/opennms/devjam2022/apiserver/model/UserRole.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+public class UserRole {
+    String id;
+    String userIdentity;
```
Comment (yangli2014) [inline api-server/src/main/java/org/opennms/devjam2022/apiserver/model/UserRole.java]: Do we need this userIdentity inside the role?

Comment (denkonakov) [reply api-server/src/main/java/org/opennms/devjam2022/apiserver/model/UserRole.java]: Not really for our current goals. Removed. Thanks

File: api-server/src/main/java/org/opennms/devjam2022/apiserver/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+@RestController
+@RequestMapping("/v1/user")
+public class UserController {
+
+    @Qualifier("InMemoryUserService")
+    @Autowired
+    IUserService userService;
+
+    @GetMapping("/users")
+    public List<UserWithRoles> all() {
```
Comment (yangli2014) [inline api-server/src/main/java/org/opennms/devjam2022/apiserver/controller/UserController.java]: When we use Sprint webflux, it is sprint reactive and the return type will be Flux<User> for the list of users, Monon<User> for the single suer.

File: api-server/src/main/java/org/opennms/devjam2022/apiserver/model/UserWithRoles.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+public class UserWithRoles {
+    String email;
+    String identity;
```
Comment (yangli2014) [inline api-server/src/main/java/org/opennms/devjam2022/apiserver/model/UserWithRoles.java]: Use number as ID.

File: api-server/src/main/java/org/opennms/devjam2022/apiserver/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import org.opennms.devjam2022.apiserver.impl.IUserService;
+import org.opennms.devjam2022.apiserver.model.UserRole;
+import org.opennms.devjam2022.apiserver.model.UserWithRoles;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/v1/user")
```
Comment (yangli2014) [inline api-server/src/main/java/org/opennms/devjam2022/apiserver/controller/UserController.java]:
```suggestion
@RequestMapping("/v1/users")
```

File: api-server/src/main/java/org/opennms/devjam2022/apiserver/controller/UserController.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+public class UserController {
+
+    @Qualifier("InMemoryUserService")
+    @Autowired
+    IUserService userService;
+
+    @GetMapping("/users")
```
Comment (yangli2014) [inline api-server/src/main/java/org/opennms/devjam2022/apiserver/controller/UserController.java]: we can skip "/users" here

**Review by yangli2014 — COMMENTED** [review]
greate job.

File: api-server/src/main/java/org/opennms/devjam2022/apiserver/impl/InMemoryUserService.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+ */
+@Component("InMemoryUserService")
+public class InMemoryUserService implements IUserService {
+
+    private static final HashMap USER_ROLES = new HashMap<String, List<UserRole>>();
+    private static final List<UserWithRoles> USERS = new ArrayList<>();
+
+    // TECH-DEBT: This is not thread-safe first of all. Secondly, we can't delete pre-created users or roles.
+    // most probably need to be refactored later to something more sophisticated, considering that Akka may
+    // use HTTP2.0 and streaming to access this?
```
Comment (yangli2014) [inline api-server/src/main/java/org/opennms/devjam2022/apiserver/impl/InMemoryUserService.java:21]: It is OK for this project. And we can use ConcurrentMap here

File: api-server/src/main/java/org/opennms/devjam2022/apiserver/controller/UserController.java (hunk 2 above)
Comment (denkonakov) [reply api-server/src/main/java/org/opennms/devjam2022/apiserver/controller/UserController.java]: Good catch. Thanks. Refactored. Issue #11 created for the future reference

**Review by yangli2014 — APPROVED** [review]

File: api-server/src/main/java/org/opennms/devjam2022/apiserver/model/UserWithRoles.java (hunk 3 above)
Comment (denkonakov) [reply api-server/src/main/java/org/opennms/devjam2022/apiserver/model/UserWithRoles.java]: Done

File: api-server/src/main/java/org/opennms/devjam2022/apiserver/controller/UserController.java (hunk 4 above)
Comment (denkonakov) [reply api-server/src/main/java/org/opennms/devjam2022/apiserver/controller/UserController.java]: Done

File: api-server/src/main/java/org/opennms/devjam2022/apiserver/controller/UserController.java (hunk 5 above)
Comment (denkonakov) [reply api-server/src/main/java/org/opennms/devjam2022/apiserver/controller/UserController.java]: Done

**Pull request merged**